app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -711,7 +711,7 @@ define([
                 }
             );
 
-            if (result.oldPrice.amount !== result.finalPrice.amount) {
+            if (typeof result != 'undefined' && result.oldPrice.amount !== result.finalPrice.amount) {
```

Comment:
@aholovan Thanks for the background. Making the `result` always present is a good idea, but it seems to be out of scope for this PR.

@korostii I look at a fairly random sample of pull requests; my schedule doesn't allow me to review every single one. I didn't notice that this was a backport, so that's on me!

As for strict comparison, it really is a standard practice in all cases. Linters in standard configuration will complain. But it's all right; in a later project, we'll use an automated tool to change all of these loose comparisons to strict ones. So I will approve this!